CONTRIBUTING.md

```diff
@@ -1,61 +1,118 @@
+- [`flutter_plugin_tools`](https://pub.dartlang.org/packages/flutter_plugin_tools) locally activated.
+- [`tuneup`](https://pub.dev/packages/tuneup) locally activated.
 
-## Getting the code and configuring your environment
+## 2. Forking & cloning the repository
 
+- Ensure all the dependencies described in the previous section are installed.
+- Fork `https://github.com/FirebaseExtended/flutterfire` into your own GitHub account. If
+  you already have a fork, and are now installing a development environment on
+  a new machine, make sure you've updated your fork so that you don't use stale
+  configuration options from long ago.
+- If you haven't configured your machine with an SSH key that's known to github, then
+  follow [GitHub's directions](https://help.github.com/articles/generating-ssh-keys/)
+  to generate an SSH key.
+- `git clone git@github.com:<your_name_here>/flutterfire.git`
+- `git remote add upstream git@github.com:FirebaseExtended/flutterfire.git` (So that you
+  fetch from the master repository, not your clone, when running `git fetch`
+  et al.)
 
- * Ensure all the dependencies described in the previous section are installed.
- * Fork `https://github.com/FirebaseExtended/flutterfire` into your own GitHub account. If
-   you already have a fork, and are now installing a development environment on
-   a new machine, make sure you've updated your fork so that you don't use stale
-   configuration options from long ago.
- * If you haven't configured your machine with an SSH key that's known to github, then
-   follow [GitHub's directions](https://help.github.com/articles/generating-ssh-keys/)
-   to generate an SSH key.
- * `git clone git@github.com:<your_name_here>/flutterfire.git`
- * `cd plugins`
- * `git remote add upstream git@github.com:FirebaseExtended/flutterfire.git` (So that you
-   fetch from the master repository, not your clone, when running `git fetch`
-   et al.)
+## 3. Environment Setup
 
-## Running the examples
+FlutterFire uses [Melos](https://github.com/invertase/melos) to manage the project and dependencies.
 
+To install Melos, run the following command from your SSH client:
 
-To run an example with a prebuilt binary from the cloud, switch to that
-example's directory, run `pub get` to make sure its dependencies have been
-downloaded, and use `flutter run`. Make sure you have a device connected over
-USB and debugging enabled on that device.
+```bash
+pub global activate melos
+```
+
+Next, at the root of your locally cloned repository bootstrap the projects dependencies:
+
+```bash
+melos bootstrap
+```
+
+The bootstrap command locally links all dependencies within the project without having to
+provide manual [`dependency_overrides`](https://dart.dev/tools/pub/pubspec). This allows all
+plugins, examples and tests to build from the local clone project.
+
+## 4. Running an example
```

Comment:
Could we include a note here that running `melos bootstrap` is required before running the examples?

Reply:
Mentioned in the previous step above, though I think it's worth maybe adding an additional bit of info that mentions that `pub get` inside any package in the project is not required once the project has been bootstrapped and that if dependencies have been changed you just need to run bootstrap again

Reply:
Sounds good, lets mention that!